libs/langchain_v1/langchain/agents/middleware/human_in_the_loop.py

```diff
@@ -1,135 +1,255 @@
+            action_request=ActionRequest(
+                action="run_command",  # The action being requested
+                args={"command": "ls", "args": ["-l"]},  # Arguments for the action
+            ),
+            config=HumanInTheLoopConfig(
+                allow_accept=True,  # Allow approval
+                allow_respond=True,  # Allow rejection with feedback
+                allow_edit=False,  # Don't allow approval with edits
+            ),
+            description="Please review the command before execution",
+        )
+        # Send the interrupt request and get the response
+        response = interrupt([request])[0]
+        ```
+    """
+
+    action_request: ActionRequest
+    config: HumanInTheLoopConfig
+    description: str | None
+
+
+class AcceptPayload(TypedDict):
+    """Response when a human approves the action."""
+
+    type: Literal["accept"]
+
+
+class ResponsePayload(TypedDict):
+    """Response when a human rejects the action."""
+
+    type: Literal["response"]
+
+    args: NotRequired[str]
+
+
+class EditPayload(TypedDict):
+    """Response when a human edits the action."""
+
+    type: Literal["edit"]
+
+    args: ActionRequest
+
+
+HumanInTheLoopResponse = AcceptPayload | ResponsePayload | EditPayload
+
+
+class ToolConfig(TypedDict):
+    """Configuration for a tool requiring human in the loop.
+
+    Attributes:
+        allow_accept: Whether the human can approve the current action without changes
+        allow_edit: Whether the human can approve the current action with edited content
+        allow_respond: Whether the human can reject the current action with feedback
+        description: The description attached to the request for human input
+    """
+
+    allow_accept: NotRequired[bool]
+    allow_edit: NotRequired[bool]
+    allow_respond: NotRequired[bool]
+    description: NotRequired[str]
 
 
 class HumanInTheLoopMiddleware(AgentMiddleware):
     """Human in the loop middleware."""
 
     def __init__(
         self,
-        tool_configs: ToolInterruptConfig,
-        message_prefix: str = "Tool execution requires approval",
+        tool_configs: dict[str, bool | ToolConfig],
+        *,
+        description_prefix: str = "Tool execution requires approval",
     ) -> None:
         """Initialize the human in the loop middleware.
 
         Args:
-            tool_configs: The tool interrupt configs to use for the middleware.
-            message_prefix: The message prefix to use when constructing interrupt content.
+            tool_configs: Mapping of tool name to allowed actions.
+                If a tool doesn't have an entry, it's auto-approved by default.
+                * `True` indicates all actions are allowed: accept, edit, and respond.
+                * `False` indicates that the tool is auto-approved.
+                * ToolConfig indicates the specific actions allowed for this tool.
+            description_prefix: The prefix to use when constructing action requests.
```

Comment:
Could we specify that the description prefix does not apply to tools that have a description manually specified?